tests/validators/test_model_fields.py

```diff
@@ -347,8 +347,8 @@ def test_validate_assignment_allow_extra():
     assert v.validate_python({'field_a': 'test'}) == ({'field_a': 'test'}, {}, {'field_a'})
 
     assert v.validate_assignment({'field_a': 'test'}, 'other_field', 456) == (
-        {'field_a': 'test', 'other_field': 456},
-        None,
+        {'field_a': 'test'},
+        {'other_field': 456},
         {'other_field'},
     )
```

Comment:
Based on this it looks like the type annotation in `_pydantic_core.pyi` needs to be updated to a 3-tuple.